### PR TITLE
Feature/update properties

### PR DESCRIPTION
**Pull request by yaojiefeng**
Full diff can be found: https://issues.cask.co/browse/CDAP-15383
I didn't add preview config since I don't think we are starting preview as master service in OSS.

### CODE REVIEW

File: src/descriptor/service.sdl [hunk 1]
```diff
@@ -2673,6 +2818,25 @@
...
+  	"name": "log_query_server_bind_address",
+  	"label": "Log Query Server Bind Address",
+  	"description": "Log query HTTP service bind address",
+  	"configName": "log.query.server.bind.address",
+  	"configurableInWizard": false,
+  	"default": "0.0.0.0",
+  	"type": "string"
+  },
+  {
+    "name": "log_query_server_bind_port",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: indentation

File: src/descriptor/service.sdl [hunk 2]
```diff
@@ -2567,6 +2703,15 @@
 	  "type": "long",
 	  "unit": "bytes"
 	},
+	{
+  	"name": "log_pipeline_cdap_file_cleanup_batch_size",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: indentation

File: src/descriptor/service.sdl [hunk 3]
```diff
@@ -914,6 +922,15 @@
       "configurableInWizard": false,
       "type": "string"
     },
+    {
+      "name": "master_environment_extensions_dir",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: this one probably should go under the master component properties (same as runtime_extensions_dir).

File: src/descriptor/service.sdl [hunk 4]
```diff
@@ -1081,14 +1098,99 @@
       "unit": "megabytes"
     },
     {
-      "name": "metrics_data_table_retention_resolution_1_seconds",
-      "label": "Metrics Data Table Retention Resolution 1 Seconds",
-      "description": "Retention resolution in seconds of the 1-second resolution table; default retention period is 2 hours",
-      "configName": "metrics.data.table.retention.resolution.1.seconds",
+      "name": "messaging_http_server_bind_address",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: assuming this pertains to the system container, these bind address/port settings should probably be moved under the master component properties, which is where the other system container bind properties are

File: src/descriptor/service.sdl [hunk 5]
```diff
@@ -272,6 +272,67 @@
...
+      "configurableInWizard": false,
+      "default": 800,
+      "type": "long"
+    },
+    {
+      "name": "data_tx_enabled",
+      "label": "Data Tx Enabled",
+      "description": "Determines if the transaction service is enabled.",
+      "configName": "data.tx.enabled",
+      "configurableInWizard": true,
```
[the remark this answers is not in the document]
Comment (yaojiefeng) [reply src/descriptor/service.sdl]: 👍

File: src/descriptor/service.sdl [hunk 6]
```diff
@@ -1081,14 +1098,99 @@
...
+    },
+    {
+      "name": "messaging_http_server_bind_port",
+      "label": "Messaging Http Server Bind Port",
+      "description": "Messaging HTTP service bind port; if 0, binds to a random port",
+      "configName": "messaging.http.server.bind.port",
+      "configurableInWizard": false,
+      "default": 0,
+      "type": "long",
+      "max": 65536
```
[the remark this answers is not in the document]
Comment (yaojiefeng) [reply src/descriptor/service.sdl]: 👍

File: src/descriptor/service.sdl [hunk 7]
```diff
@@ -2517,6 +2598,61 @@
...
+          "type": "long",
+        },
+        {
+          "name": "log_queue_size",
+          "label": "Log Queue Size",
+          "description": "The buffer size used in Log Appender",
+          "configName": "log.queue.size",
+          "configurableInWizard": false,
+          "default": 512,
+          "type": "long",
```
[the remark this answers is not in the document]
Comment (yaojiefeng) [reply src/descriptor/service.sdl]: This is the number of log messages, so it is just a number, it is not in unit bytes or mb

File: src/descriptor/service.sdl [hunk 8]
```diff
@@ -1081,14 +1089,62 @@
...
-      "label": "Metrics Data Table Retention Resolution 1 Seconds",
-      "description": "Retention resolution in seconds of the 1-second resolution table; default retention period is 2 hours",
-      "configName": "metrics.data.table.retention.resolution.1.seconds",
+      "name": "metrics_admin_topic",
+      "label": "Metrics Admin Topic",
+      "description": "Topic name for publishing metrics administrative events to the messaging system",
+      "configName": "metrics.admin.topic",
+      "configurableInWizard": false,
+      "default": "metricsadmin",
+      "type": "string",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: extra comma, here and in several other places.  make sure it's valid json before submitting.  (the bamboo build also will run a validation check)

File: src/descriptor/service.sdl [hunk 9]
```diff
@@ -1487,6 +1533,18 @@
...
+    {
+      "name": "security_store_extensions_dir",
+      "label": "Security Store Extensions Dir",
+      "description": "Semicolon-separated list of local directories that are scanned for secure store extensions.",
+      "configName": "security.store.extensions.dir",
+      "configurableInWizard": false,
+      "default": [
+      	"${cdap.home}/master/ext/securestores"
+      ],
+   	  "type": "string_array",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: indentation

File: src/descriptor/service.sdl (hunk 7 above)
[the remark this answers is not in the document]
Comment (dereklwood) [reply src/descriptor/service.sdl]: ok, we have a peculiar choice of default value then :)

File: src/descriptor/service.sdl [hunk 10]
```diff
@@ -2816,15 +2928,75 @@
...
+          "description": "Metrics query HTTP service bind address",
+          "configName": "metrics.bind.address",
+          "configurableInWizard": false,
+          "default": "0.0.0.0",
+          "type": "string",
+        },
+        {
+          "name": "metrics_bind_port",
+          "label": "Metrics Bind Port",
+          "description": "Metrics query HTTP service bind port; if 0, binds to a random port ",
```
Comment (dereklwood) [inline src/descriptor/service.sdl]: extra space after `port`

**Comment by yaojiefeng** [comment]
Validated the file is valid json, merging

**Pull request merged**